marathon/marathon.go

```diff
@@ -243,26 +243,26 @@ func (m Marathon) urlWithQuery(path string, params params) string {
 	return marathon.String()
 }
 
-func (m *Marathon) IsLeader() (bool, error) {
+func (m *Marathon) IsLeader(port uint32) (bool, error) {
 	if m.MyLeader == "*" {
 		log.Debug("Leader detection disable")
 		return true, nil
 	}
 	if m.MyLeader == "" {
-		if err := m.resolveHostname(); err != nil {
+		if err := m.resolveHostname(port); err != nil {
 			return false, fmt.Errorf("Could not resolve hostname: %v", err)
 		}
 	}
 	leader, err := m.Leader()
 	return m.MyLeader == leader, err
 }
 
-func (m *Marathon) resolveHostname() error {
+func (m *Marathon) resolveHostname(port uint32) error {
 	hostname, err := hostname()
 	if err != nil {
 		return err
 	}
-	m.MyLeader = fmt.Sprintf("%s:8080", hostname)
```

Comment:
Instead of `8080` there should be something like this

```go
u, err := url.Parse(m.Location)
if err != nil {
   return fmt.Errorf("Could not parse marathon location (%s): %v", m.Location, err);
}
m.MyLeader = fmt.Sprintf("%s:%d", hostname, u.Port())
```